### PR TITLE
improve debuggability in jobset controller tests

### LINKED ISSUE #2227: Jobsets having replicas = 0 for their replicatedJob are not handled in Kueue
**What happened**:
- Created a jobset with two replicatedJobs: the first one had a replica count of 1 and the other one had a replica count of 0.
- The jobset was created in a suspended state and had the required kueue labels/annotations
- The jobset was NOT unsuspended by Kueue and the 1 expected job did not start

**What you expected to happen**:
- Kueue should have unsuspended the jobset and started the job

**How to reproduce it (as minimally and precisely as possible)**:
- As mentioned above, create a jobset was two replicatedJobs: one with replicas > 0 and another with replicas = 0

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): `v1.28.8`
- Kueue version (use `git describe --tags --dirty --always`):`v0.6`
- Cloud provider or hardware configuration: `aws`
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

** Details **

Jobset CRD shows the default is 1 but it is not a minimum value of 1. So 0 is acceptable.
```
                    replicas:
                      default: 1
                      description: |-
                        Replicas is the number of jobs that will be created from this ReplicatedJob's template.
                        Jobs names will be in the format: <jobSet.name>-<spec.replicatedJob.name>-<job-index>
                      format: int32
                      type: integer
                    template:
```

Kueue's workload CRD shows that the podSet's count has a minimum value of 1:
```
                  properties:
                    count:
                      description: count is the number of pods for the spec.
                      format: int32
                      minimum: 1
                      type: integer
                    minCount:
                      description: |-
                        minCount is the minimum number of pods for the spec acceptable
                        if the workload supports partial admission.


                        If not provided, partial admission for the current PodSet is not
                        enabled.


                        Only one podSet within the workload can use this.


                        This is an alpha field and requires enabling PartialAdmission feature gate.
                      format: int32
                      minimum: 1
                      type: integer
```

Kueue's log show the following error:
```
{"level":"error","ts":"2024-05-17T20:59:35.924123528Z","caller":"controller/controller.go:329","msg":"Reconciler error","controller":"jobset","controllerGroup":"jobset.x-k8s.io","controllerKind":"JobSet","JobSet":{"name":"js-b8b2dc","namespace":"jobs-default"},"namespace":"jobs-default","name":"js-b8b2dc","reconcileID":"9fbf19e9-35a9-443a-a60d-78ef69dadaba","error":"Workload.kueue.x-k8s.io "jobset-js-b8b2dc-9431d" is invalid: spec.podSets[1].count: Invalid value: 0: spec.podSets[1].count in body should be greater than or equal to 1","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.17.0/pkg/internal/controller/controller.go:227"}
```

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Improve debuggability of jobset controller tests as requested [here](https://github.com/kubernetes-sigs/kueue/pull/2268#pullrequestreview-2085007373)
 
#### Which issue(s) this PR fixes:

Fixes #2227

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by mszadkow** [comment]
@tenzen-y please review, I followed your comments to pass `gomega.Gomega` to other `Eventually` in the same file

File: test/integration/controller/jobs/jobset/jobset_controller_test.go [hunk 1]
```diff
@@ -271,12 +254,11 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
...
 			util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, createdWorkload)
-			gomega.Eventually(func() bool {
-				if err := k8sClient.Get(ctx, lookupKey, createdJobSet); err != nil {
-					return false
-				}
-				return !*createdJobSet.Spec.Suspend
-			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, lookupKey, createdJobSet)).Should(gomega.Succeed())
+				g.Expect(createdJobSet.Spec.Suspend).ShouldNot(gomega.BeNil())
```
Comment (mimowo) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:259]: This check does not look necessary given the check below

File: test/integration/controller/jobs/jobset/jobset_controller_test.go [hunk 2]
```diff
@@ -323,11 +299,12 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
...
 			ginkgo.By("create the jobset", func() {
 				gomega.Expect(k8sClient.Create(ctx, mixJobSet)).Should(gomega.Succeed())
-				gomega.Eventually(func() *bool {
+				gomega.Eventually(func(g gomega.Gomega) {
 					createdJobSet1 := &jobsetapi.JobSet{}
-					gomega.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
-					return createdJobSet1.Spec.Suspend
-				}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
+					g.Expect(k8sClient.Get(ctx, jobSetLookupKey, createdJobSet1)).Should(gomega.Succeed())
+					g.Expect(createdJobSet1.Spec.Suspend).ShouldNot(gomega.BeNil())
```
Comment (mimowo) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:305]: ditto

File: test/integration/controller/jobs/jobset/jobset_controller_test.go [hunk 3]
```diff
@@ -441,10 +418,11 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
 			})
 
 			ginkgo.By("fetch the job and verify it is suspended as the checks are not ready", func() {
-				gomega.Eventually(func() *bool {
-					gomega.Expect(k8sClient.Get(ctx, *jobLookupKey, createdJob)).Should(gomega.Succeed())
-					return createdJob.Spec.Suspend
-				}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, *jobLookupKey, createdJob)).Should(gomega.Succeed())
+					g.Expect(createdJob.Spec.Suspend).ShouldNot(gomega.BeNil())
```
Comment (mimowo) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:423]: ditto

File: test/integration/controller/jobs/jobset/jobset_controller_test.go [hunk 4]
```diff
@@ -522,10 +500,11 @@ var _ = ginkgo.Describe("JobSet controller", ginkgo.Ordered, ginkgo.ContinueOnFa
 			})
 
 			ginkgo.By("await for the job to be admitted", func() {
-				gomega.Eventually(func() *bool {
-					gomega.Expect(k8sClient.Get(ctx, *jobLookupKey, createdJob)).Should(gomega.Succeed())
-					return createdJob.Spec.Suspend
-				}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(false)))
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, *jobLookupKey, createdJob)).Should(gomega.Succeed())
+					g.Expect(createdJob.Spec.Suspend).ShouldNot(gomega.BeNil())
```
Comment (mimowo) [inline test/integration/controller/jobs/jobset/jobset_controller_test.go:505]: ditto

**Review by mimowo — COMMENTED** [review]
LGTM, just the nit about the unnecessary check

**Review by tenzen-y — COMMENTED** [review]
Thank you.
/approve

After you address @mimowo 's comments, we can merge this PR.

**Review by tenzen-y — COMMENTED** [review]
Great contribution!
Thank you!
/lgtm

**Pull request merged**